knowhere/index/vector_index/IndexHNSW.cpp

```diff
@@ -252,7 +252,7 @@ IndexHNSW::QueryImpl(int64_t n,
         feder = std::make_unique<feder::hnsw::FederResult>();
     }
 
-    auto dim = GetMetaDim(config);
+    auto dim = index_->space_->get_data_size() / sizeof(float);
```

Comment:
Why we don't pass dim in from `Query()` ?

Reply:
It's better to get dim from index

Reply:
Then we have three place to get dim:
1. From `DataSet` passed in.
2. From `Dim()` function in `IndexHNSW`
3. From `index_->space_->get_data_size() / sizeof(float)`

this is too confusing, can we ensure they are consistent and which one should we pick?